lxd/db/generate/db/parse.go

```diff
@@ -139,12 +139,48 @@ func Parse(pkg *ast.Package, name string, kind string) (*Mapping, error) {
 		return nil, errors.Wrapf(err, "Failed to parse %q", name)
 	}
 
+	filterStr := findStruct(pkg.Scope, name+"Filter")
+	if filterStr == nil {
+		return nil, fmt.Errorf("No declaration found for %q", name+"Filter")
+	}
+
+	filters, err := parseStruct(filterStr, kind)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to parse %q", name)
+	}
+
 	m := &Mapping{
 		Package: pkg.Name,
 		Name:    name,
 		Fields:  fields,
 	}
 
+	for i, filter := range filters {
+		// Filter field must be present in original struct.
+		field := m.FieldByName(filter.Name)
+		if field == nil {
+			return nil, fmt.Errorf("filter field %q is not in struct %q", filter.Name, name)
```

Comment:
Error messages should start with upper case first letter, like you've done for some of the other errors in this commit.